tools/bot/bot.go

```diff
@@ -0,0 +1,172 @@
+	message := Message{
+		Text: getFormattedMessageText(GenerateInfoText(alert)),
+		Prio: hasPrio(alert.Stream.ID, statsDao) || alert.IsLecturer,
+	}
+	return b.SendMessage(message)
+}
+
+// GenerateInfoText generates a formatted issue text, should be visible on any client that supports markdown and HTML.
+func GenerateInfoText(botInfo AlertMessage) string {
+	combIP := strings.Split(botInfo.CombIP, "/")[0] // URL has /extron[...]
+
+	var infoText string
+
+	infoText += "🚨 **Technical problem**\n\n" +
+		"<table><tr><th>Categories</th><td>" + botInfo.Categories + "</td></tr>" +
+		"<tr><th>Course name</th><td>" + botInfo.CourseName + "</td></tr>" +
+		"<tr><th>Stream URL</th><td>" + botInfo.StreamUrl + "</td></tr>" +
+		"<tr><th>Description</th><td>" + botInfo.Comment + "</td></tr>"
+
+	if !botInfo.Stream.IsSelfStream() {
+		if botInfo.LectureHall != "" {
+			infoText += "<tr><th>Lecture hall</th><td>" + botInfo.LectureHall + "</td></tr>"
+		}
+		if combIP != "" {
+			infoText += "<tr><th>Combined IP</th><td>" + combIP + "</td></tr>"
+		}
+		if botInfo.CameraIP != "" {
+			infoText += "<tr><th>Camera IP</th><td>" + botInfo.CameraIP + "</td></tr>"
+		}
+	}
+	infoText += "</table>📢 **Contact information**\n\n<table>"
+	// Has the person that reported the issue entered custom contact data?
+	if botInfo.Name != "" {
+		infoText += "<tr><th>Name</th><td>" + botInfo.User.Name + "</td></tr>"
+	} else if botInfo.User.Name != "" {
+		if botInfo.User.LastName != nil {
+			infoText += "<tr><th>Name</th><td>" + botInfo.User.Name + " " + *botInfo.User.LastName + "</td></tr>"
+		} else {
+			infoText += "<tr><th>Name</th><td>" + botInfo.User.Name + "</td></tr>"
+		}
+	}
+	if botInfo.Email != "" {
+		infoText += "<tr><th>Email</th><td>" + botInfo.Email + "</td></tr>"
+	} else if botInfo.User.Email.Valid {
+		infoText += "<th>Email</th><td>" + botInfo.User.Email.String + "</td></tr>"
+	}
+	if botInfo.PhoneNumber != "" {
+		infoText += "<tr><th>Phone</th><td>" + botInfo.PhoneNumber + "</td></tr>"
+	}
+	infoText += "</table>"
+
+	return infoText
+}
+
+// getFormattedMessageText generates a HTML styled message bot info
+func getFormattedMessageText(message string) string {
+	unsafe := blackfriday.Run([]byte(message))
+	// Sanitization already in place since we want to edit user generated content soon
+	html := bluemonday.
+		UGCPolicy().
+		AddTargetBlankToFullyQualifiedLinks(true).
+		SanitizeBytes(unsafe)
+	return string(html)
+}
+
+// hasPrio returns true if 1% of the current viewers of a stream with streamID reported an issue.
+// When there threshold for sending an alert is greater than 1, it is also checked whether these reports are consecutive.
+func hasPrio(streamID uint, statsDao dao.StatisticsDao) bool {
+	liveViewers, err := statsDao.GetCurrentLiveViewers(streamID)
+	if err != nil {
+		sentry.CaptureException(err)
+		log.WithError(err).Error("Failed to get current live viewers")
+		return false
+	}
+	distinctReports := liveViewers
```

Comment:
why is the number of reports initialized as the number of live viewers instead of `len(issuesPerStream[streamID])`? This might need a comment.

Reply:
Yeah, I think that was just some last minute refactoring mistake, it should be `len(issuesPerStream[streamID])` of course